linker/linker_test.go

```diff
@@ -3290,17 +3369,17 @@ func TestLinkerValidation(t *testing.T) {
 				if tc.expectedDiffWithProtoc {
 					// We can explicitly check different result is produced by protoc. When the bug is fixed,
 					// we can change the tc.expectedDiffWithProtoc field to false and delete the comment.
-					require.False(t, passProtoc)
+					require.False(t, passProtoc, "expected protoc to disallow the case, but it allows it")
 				} else {
 					// if the test case passes protocompile, it should also pass protoc.
-					require.True(t, passProtoc)
+					require.True(t, passProtoc, "protoc should allow the case")
 				}
 			} else {
 				if tc.expectedDiffWithProtoc {
-					require.True(t, passProtoc)
+					require.True(t, passProtoc, "expected protoc to allow the case, but it disallows it")
 				} else {
 					// if the test case fails protocompile, it should also fail protoc.
-					require.False(t, passProtoc)
+					require.False(t, passProtoc, "protoc should disallow the case")
```

Comment:
Without this message, it was a bit inscrutable if the test failed because protoc's behavior was unexpected: it was basically "should be true but was false" or vice versa (or something similar... whatever the default message is with `require.True/False`).

There are two different "forms" for the message now intentionally:
1. The "protoc should ..." is asserting that protoc **should** behave this way, and it would match the way protocompile behaves. Full stop.
2. The "expected protoc to ..." is when we expect protoc to behave a particular way, but not because it necessarily "should" but because its behavior is known to differ from protocompile (typically due to a bug or other limitation in protoc that does not exist in protocompile).

Hopefully that makes sense.